doc/guides/dts/design.rst

```diff
@@ -78,9 +78,8 @@ Example remaining work
   dtschema does not yet meet Zephyr's needs, we should try to follow what it is
   capable of representing in Zephyr's own bindings.
 
-- Due to namespace collisions and inflexibility in the bindings language,
-  Zephyr cannot support the full set of possible node paths and bindings
-  supported by Linux.
+- Due to inflexibility in the bindings language, Zephyr cannot support the full
+  set of bindings supported by Linux.
```

Comment:
what is this trying to cover?  Not issue with the intent, just not exactly sure what case we have issues w/from a binding point of view.

Reply:
Linux is more general than Zephyr when handling specifier spaces and nexus nodes at least.

For example, our handling of `#foo-cells` is done by just stripping the trailing `-s` from `foos`:

https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/dts/edtlib.py#L1417

While convenient and though it gets the job done for now, it's not sufficiently expressive for the `mboxes` / `#mbox-cells` binding for example:

https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/mailbox/mailbox.txt#L9

Our handling of the "compound" type (i.e. generate nothing) is another example that holds us back from expressing nexus nodes with the same level of generality as Linux, IIUC.